client/a8c-for-agencies/sections/sites/site-actions/remove-site-action-modal.tsx

```diff
@@ -0,0 +1,92 @@
+import { Button, __experimentalHStack as HStack } from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
+import useRemoveSiteMutation from 'calypso/a8c-for-agencies/data/sites/use-remove-site';
+import { useDispatch } from 'calypso/state';
+import { successNotice } from 'calypso/state/notices/actions';
+import type { SiteData } from '../../../../jetpack-cloud/sections/agency-dashboard/sites-overview/types';
+
+const createRemoveSiteActionModal =
+	( {
+		onRefetchSite,
+		recordTracksEventRemoveSite,
+	}: {
+		onRefetchSite?: () => Promise< unknown >;
+		recordTracksEventRemoveSite: () => void;
+	} ) =>
+	( {
+		items,
+		closeModal,
+		onActionPerformed,
+	}: {
+		items: SiteData[];
+		closeModal: () => void;
+		onActionPerformed?: () => void;
+	} ) => {
+		const translate = useTranslate();
+		const dispatch = useDispatch();
+		const { mutate: removeSite } = useRemoveSiteMutation();
+
+		recordTracksEventRemoveSite();
```

Comment:
The event should only be recorded if the user effectively confirms the removal, not if the user cancels as happens now.

Reply:
That's good thinking. However, note this is working as it was [before](https://github.com/Automattic/wp-calypso/blob/add/actions-to-a4a-sites/client/a8c-for-agencies/sections/sites/site-actions/use-site-actions.ts#L79). The event tracks that the action was clicked (and the modal opened for this event).